### PR TITLE
Rename 'Solution Events' to 'Legacy Solution Events'

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Added docs as well to indicate this is a transitional API, and should not be used for new features.

I also simplified the data model here.  I realized i didn't need to transform this into solution-crawler concepts.  I could just just keep this close to the existing WorkspaceEvents concept we already have (And which solution-crawler wants to consume.  We're just now effectively providing a stream of those raw events directly to OOP for this legacy case for the new owners of the solution crawler to consume until they move off onto their own pull model.

### CODE REVIEW

File: src/Features/Core/Portable/LegacySolutionEvents/ILegacySolutionEventsAggregationService.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    /// <summary>
+    /// This is a legacy api intended only for existing SolutionCrawler partners to continue to function (albeit with
+    /// ownership of that crawling task now belonging to the partner team, not roslyn).  It should not be used for any
+    /// new services.
+    /// </summary>
+    internal interface ILegacySolutionEventsAggregationService : IWorkspaceService
+    {
+        ValueTask OnWorkspaceChangedEventAsync(WorkspaceChangeEventArgs args, CancellationToken cancellationToken);
+        ValueTask OnTextDocumentOpenedAsync(TextDocumentEventArgs args, CancellationToken cancellationToken);
+        ValueTask OnTextDocumentClosedAsync(TextDocumentEventArgs args, CancellationToken cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/LegacySolutionEvents/ILegacySolutionEventsAggregationService.cs:25]: only 3 events now (The same 3 that WorkCoordinator listens for).

File: src/EditorFeatures/Core/SolutionEvents/HostLegacySolutionEventsWorkspaceEventListener.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LegacySolutionEvents
+{
+    [ExportEventListener(WellKnownEventListeners.Workspace, WorkspaceKind.Host), Shared]
+    internal sealed partial class HostLegacySolutionEventsWorkspaceEventListener : IEventListener<object>
```
Comment (tmat) [inline src/EditorFeatures/Core/SolutionEvents/HostLegacySolutionEventsWorkspaceEventListener.cs:23]: >HostLegacySolutionEventsWorkspaceEventListener [](http://example.com/codeflow?start=34&length=46)

Nit: perhaps LegacyHost... would be better

File: src/Workspaces/Remote/Core/RemoteWorkspacesResources.resx [hunk 3]
```diff
@@ -213,7 +213,7 @@
   <data name="FeatureName_StackTraceExplorer" xml:space="preserve">
     <value>Stack Trace Explorer</value>
   </data>
-  <data name="FeatureName_SolutionEventsAggregation" xml:space="preserve">
-    <value>Solution Events</value>
+  <data name="FeatureName_LegacySolutionEventsAggregation" xml:space="preserve">
+    <value>Legacy Solution Events</value>
```
Comment (tmat) [inline src/Workspaces/Remote/Core/RemoteWorkspacesResources.resx:217]: >Legacy [](http://example.com/codeflow?start=11&length=6)

I wouldn't include "legacy" in the display name. Don't think we need to advertise obsoleteness to customers, if they hit gold bar caused by this service.

Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/Core/RemoteWorkspacesResources.resx:217]: sgtm.

File: src/EditorFeatures/Core/SolutionEvents/HostLegacySolutionEventsWorkspaceEventListener.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/SolutionEvents/HostLegacySolutionEventsWorkspaceEventListener.cs:23]: actually, i like keeping LegacySolutionEvents together as 'unit' of naming :)

**Comment by CyrusNajmabadi** [comment]
@shyamnamboodiripad to keep you up to date on what we're doing here.

**Pull request merged**